crates/pipeline-manager/src/api/endpoints/cluster_healthz.rs

```diff
@@ -0,0 +1,27 @@
+use actix_web::{get, web::Data as WebData, HttpResponse, Responder};
+
+use crate::api::main::ServerState;
+
+#[utoipa::path(
+    get,
+    path = "/cluster_healthz",
+    responses(
+        (status = 200, description = "All services healthy", body = HealthStatus),
+        (status = 503, description = "One or more services unhealthy", body = HealthStatus)
+    )
+)]
+#[get("/cluster_healthz")]
+pub(crate) async fn get_health(state: WebData<ServerState>) -> impl Responder {
+    let status_opt = state.health_check.read().await;
+    match &*status_opt {
+        Some(status) => {
+            if status.runner.healthy && status.compiler.healthy {
+                HttpResponse::Ok().json(status)
+            } else {
+                HttpResponse::ServiceUnavailable().json(status)
+            }
+        }
+        None => HttpResponse::ServiceUnavailable()
+            .json(serde_json::json!({"error": "Health status not available"})),
```

Comment:
Can this be an error rather than free form JSON?

Reply:
it can be ig, but this was done to have consistent body format ( JSON ) irrespective of response.

Reply:
What what is it consistent with? The OpenAPI spec?

Reply:
Yes, but a JSON body which is not what the OpenAPI spec states (e.g., HealthStatus compatible)